realsense_camera/src/base_nodelet.cpp

```diff
@@ -246,23 +246,30 @@ namespace realsense_camera
    */
   void BaseNodelet::advertiseTopics()
   {
-    image_transport::ImageTransport image_transport(nh_);
-    camera_publisher_[RS_STREAM_COLOR] = image_transport.advertiseCamera(COLOR_TOPIC, 1);
-    camera_publisher_[RS_STREAM_DEPTH] = image_transport.advertiseCamera(DEPTH_TOPIC, 1);
-    camera_publisher_[RS_STREAM_INFRARED] = image_transport.advertiseCamera(IR_TOPIC, 1);
-    pointcloud_publisher_ = nh_.advertise<sensor_msgs::PointCloud2>(PC_TOPIC, 1);
+    ros::NodeHandle color_nh(nh_, COLOR_NAMESPACE);
+    image_transport::ImageTransport color_image_transport(color_nh);
+    camera_publisher_[RS_STREAM_COLOR] = color_image_transport.advertiseCamera(COLOR_TOPIC, 1);
+
+    ros::NodeHandle depth_nh(nh_, DEPTH_NAMESPACE);
+    image_transport::ImageTransport depth_image_transport(depth_nh);
+    camera_publisher_[RS_STREAM_DEPTH] = depth_image_transport.advertiseCamera(DEPTH_TOPIC, 1);
+    pointcloud_publisher_ = depth_nh.advertise<sensor_msgs::PointCloud2>(PC_TOPIC, 1);
+
+    ros::NodeHandle ir_nh(nh_, IR_NAMESPACE);
+    image_transport::ImageTransport ir_image_transport(ir_nh);
+    camera_publisher_[RS_STREAM_INFRARED] = ir_image_transport.advertiseCamera(IR_TOPIC, 1);
   }
 
   /*
    * Advertise services.
    */
   void BaseNodelet::advertiseServices()
   {
-    get_options_service_ = nh_.advertiseService(SETTINGS_SERVICE, &BaseNodelet::getCameraOptionValues, this);
-    set_power_service_ = nh_.advertiseService(CAMERA_SET_POWER_SERVICE, &BaseNodelet::setPowerCameraService, this);
-    force_power_service_ = nh_.advertiseService(CAMERA_FORCE_POWER_SERVICE, &BaseNodelet::forcePowerCameraService, this);
-    is_powered_service_ = nh_.advertiseService(CAMERA_IS_POWERED_SERVICE, &BaseNodelet::isPoweredCameraService, this);
-}
+    get_options_service_ = pnh_.advertiseService(SETTINGS_SERVICE, &BaseNodelet::getCameraOptionValues, this);
+    set_power_service_ = pnh_.advertiseService(CAMERA_SET_POWER_SERVICE, &BaseNodelet::setPowerCameraService, this);
```

Comment:
Certainly you can setup services on a namespace. You should use a NodeHandle with that scope "camera"  to support appropriate remapping using the system tools. And then you would advertise the topics or services from the NodeHandle

As @kevincwells [pointed out](https://github.com/intel-ros/realsense/pull/117#r79445964) you want to differentiate driver API from interface API. So if it's a generic API such as `set_camera_info` it should be advertised in the `camera` namespace, but if it's an R200 implementation specific option then it is good to put it into a different namespace.

With this you could launch two drivers, remapping driver1 and driver2 to and both publishing to `camera` namespace but using `camera/driver1` and `camera/driver2` for their private namespaces. And conversely you could have one driver publishing two types of camera interfaces `camera1` and `camera2` and the driver could use the `multi_camera_driver` name.

I don't think we really have a standard for where the node implementation lives compared to the advertised namespace when they are different. Usually the NodeHandle and the camera namespace align and share the remapping.